lib/signs/utilities/messages.ex

```diff
@@ -23,252 +26,291 @@ defmodule Signs.Utilities.Messages do
+  end
+
+  defp early_am_message(current_time, scheduled_time, config) do
+    if in_early_am?(current_time, scheduled_time) do
+      %Content.Message.EarlyAm.DestinationScheduledTime{
+        destination: config.headway_destination,
+        scheduled_time: scheduled_time
+      }
     end
   end
 
-  @spec get_alert_messages(
-          Engine.Alerts.Fetcher.stop_status()
-          | {Engine.Alerts.Fetcher.stop_status(), Engine.Alerts.Fetcher.stop_status()},
-          Signs.Realtime.t()
-        ) ::
-          Signs.Realtime.sign_messages() | nil
-  defp get_alert_messages({top_alert_status, bottom_alert_status}, sign) do
-    top_alert_status
-    |> Engine.Alerts.Fetcher.higher_priority_status(bottom_alert_status)
-    |> get_alert_messages(sign)
+  defp early_am_messages(current_time, scheduled_time, config) do
+    if in_early_am?(current_time, scheduled_time) do
+      {%Content.Message.EarlyAm.DestinationTrain{destination: config.headway_destination},
+       %Content.Message.EarlyAm.ScheduledTime{scheduled_time: scheduled_time}}
+    end
   end
 
-  defp get_alert_messages(alert_status, %{pa_ess_loc: "GUNS"}) do
+  defp alert_messages(alert_status, %{pa_ess_loc: "GUNS"}, _) do
     if alert_status in [:none, :alert_along_route],
       do: nil,
       else: {%Alert.NoService{}, %Alert.UseRoutes{}}
   end
 
-  defp get_alert_messages(alert_status, sign) do
-    route = Signs.Utilities.SourceConfig.single_route(sign.source_config)
+  defp alert_messages(alert_status, sign, config) do
+    route = Signs.Utilities.SourceConfig.single_route(config)
 
     case {alert_status, sign.uses_shuttles} do
       {:shuttles_transfer_station, _} ->
-        {Content.Message.Empty.new(), Content.Message.Empty.new()}
+        {%Content.Message.Empty{}, %Content.Message.Empty{}}
 
       {:shuttles_closed_station, true} ->
         {%Alert.NoService{route: route}, %Alert.UseShuttleBus{}}
 
       {:shuttles_closed_station, false} ->
-        {%Alert.NoService{route: route}, Content.Message.Empty.new()}
+        {%Alert.NoService{route: route}, %Content.Message.Empty{}}
 
       {:suspension_transfer_station, _} ->
-        {Content.Message.Empty.new(), Content.Message.Empty.new()}
+        {%Content.Message.Empty{}, %Content.Message.Empty{}}
 
       {:suspension_closed_station, _} ->
-        {%Alert.NoService{route: route}, Content.Message.Empty.new()}
+        {%Alert.NoService{route: route}, %Content.Message.Empty{}}
 
       {:station_closure, _} ->
-        {%Alert.NoService{route: route}, Content.Message.Empty.new()}
+        {%Alert.NoService{route: route}, %Content.Message.Empty{}}
 
       _ ->
         nil
     end
   end
 
-  defp get_paging_alert_message(alert_status, uses_shuttles, destination) do
-    case {alert_status, uses_shuttles} do
-      {:shuttles_transfer_station, _} ->
-        Content.Message.Empty.new()
+  defp alert_message(alert_status, sign, config) do
+    route = Signs.Utilities.SourceConfig.single_route(config)
```

Comment:
Just making sure, but we're passing `route` along to these structs with the intention that we'll start using it in a later change right? I'm assuming that's what was being referred to in #823